source/magnifier/commands.py

```diff
@@ -0,0 +1,185 @@
+			).format(zoomLevel=defaultZoomLevel, filter=defaultFilter.name.lower())
+		)
+
+
+def zoomIn():
+	"""Zoom in the magnifier."""
+	magnifier: Magnifier = getMagnifier()
+	if magnifierIsActiveVerify(magnifier, "trying to zoom in"):
+		magnifier._zoom(True)
+		if shouldSaveShortcutChanges():
+			setDefaultZoomLevel(magnifier.zoomLevel)
+		ui.message(
+			_(
+				# Translators: Message announced when zooming in with {zoomLevel} being the target zoom level
+				"Zooming in with {zoomLevel} level"
+			).format(zoomLevel=magnifier.zoomLevel)
+		)
+
+
+def zoomOut():
+	"""Zoom out the magnifier."""
+	magnifier: Magnifier = getMagnifier()
+	if magnifierIsActiveVerify(magnifier, "trying to zoom out"):
+		magnifier._zoom(False)
+		if shouldSaveShortcutChanges():
+			# Save to config if option is enabled
+			setDefaultZoomLevel(magnifier.zoomLevel)
+		ui.message(
+			_(
+				# Translators: Message announced when zooming out with {zoomLevel} being the target zoom level
+				"Zooming out with {zoomLevel} level"
+			).format(zoomLevel=magnifier.zoomLevel)
+		)
+
+
+def toggleFilter():
+	magnifier: Magnifier = getMagnifier()
+	log.info(f"Toggling filter for magnifier: {magnifier}")
+	if magnifierIsActiveVerify(magnifier, "trying to toggle filters"):
+		filters = list(filter)
+		idx = filters.index(magnifier.filter)
+		magnifier.filter = filters[(idx + 1) % len(filters)]
+		if magnifier.magnifierType == MagnifierType.FULLSCREEN:
+			magnifier._applyFilter()
+
+		# Save to config if option is enabled
+		if shouldSaveShortcutChanges():
+			setDefaultFilter(magnifier.filter.name.lower())
+
+		ui.message(
+			_(
+				# Translators: Message announced when changing the color filter with {filter} being the new color filter
+				"Color filter changed to {filter}"
+			).format(filter=magnifier.filter.name.lower())
+		)
+
+
+def toggleFullscreenMode():
+	"""Cycle through fullscreen focus modes (center, border, relative)."""
+	magnifier: Magnifier = getMagnifier()
+	if magnifierIsActiveVerify(magnifier, "trying to change fullscreen mode"):
+		if magnifierIsFullscreenVerify(magnifier, "trying to change fullscreen mode"):
+			modes = list(FullScreenMode)
+			currentMode = magnifier.fullscreenMode
+			idx = modes.index(currentMode)
+			newMode = modes[(idx + 1) % len(modes)]
+			log.info(f"Changing fullscreen mode from {currentMode} to {newMode}")
+			magnifier.fullscreenMode = newMode
+
+			# Save to config if option is enabled
+			if shouldSaveShortcutChanges():
+				setDefaultFullscreenMode(newMode.name.lower())
+
+			ui.message(
+				_(
+					# Translators: Message announced when changing the fullscreen mode with {mode} being the new fullscreen mode
+					"Fullscreen mode changed to {mode}"
+				).format(mode=newMode.name.lower())
+			)
+
+
+def startSpotlight():
+	magnifier: FullScreenMagnifier = getMagnifier()
+	if magnifierIsActiveVerify(magnifier, "trying to start spotlight mode"):
+		if magnifierIsFullscreenVerify(magnifier, "trying to start spotlight mode"):
```

Comment:
these strings need to be translated. I suggest converting all the actions into a displayStringEnum so they are in one place